build: Fix zeromq package when cross-compiling

hebasto opened the pull request:
Since v4.3.3 (https://github.com/zeromq/libzmq/commit/068385c951c0608edec6264d55ba9c4c923acccc) `libzmq` uses `libbsd` by default.

This PR disables `libbsd` explicitly, as it's not a part of our depends. Zeromq will fallback to its internal `strlcpy` implementation.

Otherwise, on systems with installed `libbsd-dev` package the `zeromq` package build system erroneously detects `libbsd` package from the host system:

```diff
--- a/libzmq.pc
+++ b/libzmq.pc
@@ -8,5 +8,5 @@
 Version: 4.3.4
 Libs: -L${libdir} -lzmq
 Libs.private:  -lpthread
-Requires.private: 
+Requires.private:  libbsd
 Cflags: -I${includedir} 
```

This causes the `configure` fails to detect the `zeromq` package:
```
configure: WARNING: libzmq version 4.x or greater not found, disabling
```

---

Other minor improvements:
- fixed `netbsd_kevent_void.patch` offset
- disabled valgrind as it's used in unit tests which we do not run:
```diff
--- a/zmq-configure-output
+++ b/zmq-configure-output
@@ -119,11 +119,6 @@
 checking whether the g++ -m64 linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
 checking dynamic linker characteristics... (cached) GNU/Linux ld.so
 checking how to hardcode library paths into programs... immediate
-checking for valgrind... valgrind
-checking for Valgrind tool memcheck... memcheck
-checking for Valgrind tool helgrind... helgrind
-checking for Valgrind tool drd... drd
-checking for Valgrind tool exp-sgcheck... exp-sgcheck
 checking linker version script flag... --version-script
 checking if version scripts can use complex wildcards... yes
 checking for working posix_memalign... yes
```

hebasto committed:
build: Update netbsd_kevent_void.patch

This change fixes "Hunk #1 succeeded at 307 (offset -1 lines)."

hebasto commented:
Assign "23.0" milestone?

fanquake commented:
Approach NACK. Why not just pass `--disable-libbsd`? Would be good to update the PR description to actually explain what this is used for, and why it's ok to avoid.

> on some systems (impish, jammy) 

Isn't it just any system with libbsd available?

hebasto commented:
> > on some systems (impish, jammy)
> 
> Isn't it just any system with libbsd available?

Correct. The PR description updated.

hebasto commented:
> Approach NACK. Why not just pass `--disable-libbsd`? Would be good to update the PR description to actually explain what this is used for, and why it's ok to avoid.

1. It is wrong when a package build system search for its dependencies among host's packages instead of strictly relying on `$(package)_dependencies`.
2. This PR is not about "build with libbsd or avoid it". It restores behavior that was before #23956. Including libbsd in our depends build system as a `zeromq` package dependency is out of this PR scope.

fanquake commented:
> It is wrong when a package build system searches for its dependencies among host's packages instead of strictly relying on $(package)_dependencies.

I agree, but that's also something that should be solved at a higher level in the build system, rather than ad-hoc on a per-package basis. Otherwise we should just go ahead and add the same change to all other packages that use pkg-config.

> This PR is not about "build with libbsd or avoid it". It restores behavior that was before

If this PR makes a change that will ensure zeromqs build system doesn't find libbsd, how is it not about avoiding it? I'm not sure what you mean by restoring the previous behaviour (other than make things work), but I don't necessarily think that is the right goal. We're building a newer version of the package, that has a different build system and new (optional) dependencies.

> Including libbsd in our depends build system as a zeromq package dependency

I'm not suggesting we do that, especially if we end up passing `--disable-libbsd`. If we pass that flag to zeromqs config opts, it will just use it's internal `strlcpy` implementation, and libbsd will no-longer a dependency.

hebasto committed:
build: Disable libbsd when building zeromq package in depends

Since v4.3.3 (068385c951c0608edec6264d55ba9c4c923acccc) libbsd is used
by default. As we have no libbsd package in our depends, disable it
explicitly. Zeromq will fallback to its internal strlcpy implementation.

hebasto committed:
build: Disable valgrind when building zeromq package in depends

We are not running unit tests, therefore it is not required.

hebasto commented:
Updated c0f664cde99f524852b8b1fbcb7b1a938d9e2e1a -> f13e642c831c5689cb2bb7f5c4f9cb4c0c03ef21 ([pr24134.01](https://github.com/hebasto/bitcoin/commits/pr24134.01) -> [pr24134.02](https://github.com/hebasto/bitcoin/commits/pr24134.02)):

- addressed @fanquake's comments
- additionally disabled valgrind which is used in unit tests

PR description has been updated.

fanquake reviewed (approved):
ACK f13e642c831c5689cb2bb7f5c4f9cb4c0c03ef21
